src/lib/crypt.ts

```diff
@@ -4,23 +4,21 @@ import * as dotenv from "dotenv"
 
 dotenv.config()
 
-const YEAR_SECONDS = 60 * 60 * 24 * 365 // 60s * 1 hour * 24 hours * 365 days
+const YEAR_SECONDS = 60 * 60 * 24 * 365
+const isLeapYear = (year: number): boolean => {
+  return (year % 4 === 0 && year % 100 !== 0) || year % 400 === 0
+}
 
-/** Return a date in *seconds from epoch*
- * some number of (input) seconds in the future
- * This is the format JWT wants */
-export const getExpirationTime = (secondsFromNow: number): number =>
-  new Date(
-    // Expiration date is set at 10s intervals by dividing, then multiplying, by 10,000ms
-    // Also, "now" is in milliseconds and output must be seconds, so we are also dividing by 1000
-    // While we could just divide by 10,000 and multiply by 10, let's make
-    // each step explicit, so we can reason about what's happening:
-    Math.floor((new Date().valueOf() + secondsFromNow * 1000) / 10000) * 10000
+export const getExpirationTime = (secondsFromNow: number): number => {
+  const now = new Date()
+  const currentYear = now.getFullYear()
+  const yearSeconds = isLeapYear(currentYear) ? 60 * 60 * 24 * 366 : YEAR_SECONDS
```

Comment:
I really appreciate the effort you put in, here! However, `getExpirationTime()` just takes a number of seconds in the future.  The number of seconds it takes is arbitrary.  It only needs to return a number that represents "x number of seconds from now, represented as the number of seconds from Jan 1, 1970"

It is called on line 33 with `YEAR_SECONDS`, a constant that is "roughly a year hence" but the business needs of the app do not call for a precise calculation of leap year. In fact, leap year calculation introduces unnecessary complexity, which in turn can lead to errors.  For instance, line 15 `const yearSeconds` is never used!